### PR TITLE
LEARNER-4746 Test - iOS - Automate Register screen

### PR DESCRIPTION
**Pull request by nedo**
Following cases have been implemented, 

* Verify that from New Landing/New Logistration Screen, tapping "Create your Account" should load Register screen successfully
* Verify following contents are visible on screen,
** Back icon
** "Register" Title
** "Register with" label
** "Facebook" button
** "Google" button
** "or register with email" label
** Email edit-field
** "This is what you will use to login" label below
** Full Name edit-field
** "The name will be used on any certificates that you earn" label below
** Public User Name edit-field
** "The name that will identify you in your courses. It cannot be changed later." label below
** Password edit-field
** "Country or Region of Residence" spinner
** "The country or region where you live." label below
** "Show optional fields" option below
** "Create my account" button
** "By creating an account you agree to the "edX Terms of Service and Honor Code" option
* Verify all contents/elements have default value
* Verify that user should be able to scroll screen to see all contents
* Verify that tapping "Show optional fields" will turn to "Hide optional fields" and load following optional contents below,
** "Gender" spinner
** "Year of birth" spinner
** "Highest level of education completed" spinner
** "Tell us why you are interested in edX" label with edit-field below
* Verify that tapping "Hide optional fields" will turn to "Show optional fields" and all optional contents will be hidden
* Verify all optional contents/elements have default values
* Verify tapping "Back" icon will load New New Landing screen
    back to New Landing screen.
* Verify tapping "edX Terms of Service and Honor Code" loads "End User License Agreement" screen
* Verify tapping back icon from "End User License Agreement" screen
    navigate user back to Register screen.
* Verify that user is able to load EULA screen and get back to Register Screen
* Verify that user is able to load Terms screen and get back to Register Screen
* Verify that user is able to load Privacy screen and get back to Register Screen
* Verify that following input types are required,
** Email edit-field
** Full Name edit-field
** Public User Name edit-field
** Password edit-field
** "Country or Region of Residence" spinner
* Verify that app will show specific error message against each field
* Verify that tapping "Create your account" button after filling all required input(valid) types,
                will validate all inputs and load "Whats new feature screen" with specific user logged in
* Verify that tapping "Create your account" button after filling all required input(valid) types, will validate all inputs and load "Whats new feature screen" with specific user logged in
* Verify that following input types are optional,
   ** "Gender" spinner
   ** "Year of birth" spinner
   ** "Highest level of education completed" spinner
   ** "Tell us why you're interested in edX" label with edit-field below
* Verify that user should be able to log out and re-login with new created account credentials
* Update followin requirements.txt to latest libs version
** Appium-Python-Client==0.28
** pytest==3.8.0
** pytest-html==1.19.0

### CODE REVIEW

**Comment by nedo** [comment]
@raeeschachar PR is ready for review
cc @staubina

File: common/globals.py [hunk 1]
```diff
@@ -555,6 +555,42 @@ def generate_random_credentials(self, length):
...
+                                                           ))
+
+        horizontal_start_point = int(self.element_x_position)
+        vertical_start_point = int(self.element_y_position)
+        horizontal_end_point = int(self.element_x_position + (self.element_x_position * 10)/100)
+        vertical_end_point = int(self.element_y_position + (self.element_y_position * 10)/100)
+
+        coordinates = []
+        coordinates.append((horizontal_start_point, vertical_start_point))
+        coordinates.append((horizontal_end_point, vertical_end_point))
```
Comment (raeeschachar) [inline common/globals.py]: We can replace the above 3 lines with 
```
coordinates = [(horizontal_start_point, vertical_start_point), (horizontal_end_point, vertical_end_point)]
```

File: ios/tests/test_ios_register.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+from ios.pages.ios_login import IosLogin
+from ios.pages.ios_main_dashboard import IosMainDashboard
+from ios.pages.ios_new_landing import IosNewLanding
+from ios.pages.ios_register import IosRegister
+from ios.pages.ios_whats_new import IosWhatsNew
+
+
+class TestIosRegister(object):
+    """
+        Register Screen Test Case
```
Comment (raeeschachar) [inline ios/tests/test_ios_register.py]: Empty spaces in doc string

File: common/globals.py (hunk 1 above)
Comment (nedo) [reply common/globals.py]: done

File: ios/tests/test_ios_register.py (hunk 2 above)
Comment (nedo) [reply ios/tests/test_ios_register.py]: done

**Comment by nedo** [comment]
@staubina PR is ready for review

**Pull request merged**